kedro/framework/cli/starters.py

```diff
@@ -847,6 +853,10 @@ def _validate_range(start: Any, end: Any) -> None:
             message = f"'{start}-{end}' is an invalid range for project tools.\nPlease ensure range values go from smaller to larger."
             click.secho(message, fg="red", err=True)
             sys.exit(1)
+        if int(end) > max([int(num) for num in NUMBER_TO_TOOLS_NAME]):
```

Comment:
Isn't this check redundant? The purpose of _validate_range is to ensure that the range provided can be parsed into a list of numbered options. The validity of these individual options will be validated by `_validate_tools_selection`, which currently always follows `_parse_tools_input`. It feels redundant to have this check, and incomplete that we would only check for the larger number in the range, but we don't check that the smaller number is > 0. I'd opt for removing this additional condition.

Reply:
I see, could you add a comment above the condition noting that this is a safeguarding clause and not strictly part of the validation? 

As a side note, you could skip the list comprehension by using `len(NUMBER_TO_TOOLS_NAME )`

Reply:
that's right, fixed!